lib/std/terminfo.zig

```diff
@@ -0,0 +1,1068 @@
+    .{ .short = "slines", .long = .{ .string = .set_page_length } },
+    .{ .short = "smgtb", .long = .{ .string = .set_tb_margin } },
+    .{ .short = "ehhlm", .long = .{ .string = .enter_horizontal_hl_mode } },
+    .{ .short = "elhlm", .long = .{ .string = .enter_left_hl_mode } },
+    .{ .short = "elohlm", .long = .{ .string = .enter_low_hl_mode } },
+    .{ .short = "erhlm", .long = .{ .string = .enter_right_hl_mode } },
+    .{ .short = "ethlm", .long = .{ .string = .enter_top_hl_mode } },
+    .{ .short = "evhlm", .long = .{ .string = .enter_vertical_hl_mode } },
+    .{ .short = "sgr1", .long = .{ .string = .set_a_attributes } },
+    .{ .short = "slength", .long = .{ .string = .set_pglen_inch } },
+};
+
+pub fn getCapsLookUpTable(allocator: *std.mem.Allocator) !std.StringHashMap(Terms) {
+    var data = std.StringHashMap(Terms).init(allocator);
+    for (dictionary) |term| 
+        _ = try data.put(term.short, term.long);
+    return data;
+}
+
+pub const Terminfo = struct {
+    names: std.StringHashMap(bool),
+    caps: std.StringHashMap([]const u8),
+
+    pub fn init(allocator: *std.mem.Allocator, terminfo_path: []const u8, cap_delimiter: []const u8) !Terminfo {
+        var names = std.StringHashMap(bool).init(allocator);
+        var caps = std.StringHashMap([]const u8).init(allocator);
+
+        var read_line = try std.fs.ReadLineIterator.init(terminfo_path);
+        defer read_line.deinit();
+        read_line.setFilter(struct {
+            fn filter(line: []u8) bool {
+                return !(std.mem.startsWith(u8, line, "#") or line.len == 0);
+            }
+        }.filter);
+        read_line.setMap(struct {
+            fn map(line: []u8) []const u8 {
+                return std.mem.trim(u8, line[0..], " \r\t");
+            }
+        }.map);
+
+        var buffer = try read_line.collect(allocator);
+        defer buffer.deinit();
+        var tokens = std.mem.tokenize(buffer.toSlice(), cap_delimiter);
+
+        while (tokens.next()) |token| {
+            const trimmed = std.mem.trimLeft(u8, token, " ");
+
+            if (std.mem.indexOf(u8, trimmed, "|") != null) {
+                var name_tokens = std.mem.tokenize(trimmed, "|");
+                while (name_tokens.next()) |name|
+                    _ = try names.put(name, true);
+            } else if (std.mem.indexOf(u8, trimmed, "=")) |equals| {
+                _ = try caps.put(trimmed[0..equals], trimmed[equals+1..]);
+            } else if (std.mem.indexOf(u8, trimmed, "#")) |pound| {
+                _ = try caps.put(trimmed[0..pound], trimmed[pound+1..]);
+            } else {
+                _ = try caps.put(trimmed, "true");
+            }
+        }
+
+        return Terminfo {
+            .names = names,
+            .caps = caps,
+        };
+    }
+
+    pub fn deinit(self: Terminfo) void {
+        self.names.deinit();
+        self.caps.deinit();
+    }
+};
+
+test "terminfo-test" {
```

Comment:
this test should probably detect if it's on a windows machine and skip itself if it is.